content/blog/2022/02/2022-06-28-require-java-11.adoc

```diff
@@ -0,0 +1,323 @@
+
+**Beginning with Jenkins 2.357 (released on June 28, 2022) and the forthcoming September LTS release, Jenkins requires Java 11.**
+Additionally, beginning with Jenkins 2.355 (released on June 14, 2022) and Jenkins 2.346.1 LTS (released on June 22, 2022), Jenkins supports Java 17.
+Plugins have already been prepared in https://issues.jenkins.io/browse/JENKINS-68446[JENKINS-68446].
+**Use the Plugin Manager to upgrade all plugins before _and_ after upgrading to Jenkins 2.357.**
+
+Long-time Jenkins users and Java developers may recall previous migrations to Java 7 and Java 8.
+While these migrations were not trivial, they pale in comparison to the magnitude of the migration to Java 9 and beyond.
+Java 9 represents a https://en.wikipedia.org/wiki/Sea_change_(idiom)[sea change] in the Java ecosystem across many dimensions.
+Not only are there changes to licensing policies, versioning schemes, release cadences, and long-term support (LTS) cycles,
+but also there are a number of significant technical changes to the language and runtime (e.g., https://openjdk.org/projects/jigsaw/[Project Jigsaw]),
+not all of them retaining the high levels of compatibility that characterized previous releases.
+Although these evolutionary changes are likely to be in the best interests of the Java community in the long term,
+it is nevertheless incumbent on the community to be aware of the impact and to plan accordingly.
+
+In this blog post, we will discuss the Jenkins project's migration to Java 11 and Java 17,
+reflecting on it in the context of previous Java migrations within the Jenkins project and the Java community more broadly.
+We will summarize the benefits and risks and describe the upgrade process.
+Finally, we will share a glimpse at what is coming in future releases.
+
+== History
+
+Since its inception in 2005, the Jenkins (then Hudson) project has gone through a number of Java migrations.
+To a large degree, the present migration is consistent with historical precedent within the Jenkins project.
+For years the Jenkins project has published http://stats.jenkins.io[anonymous usage statistics],
+as described in https://brokenco.de/2019/05/23/jenkins-usage-stats.html[a piece by R. Tyler Croy] and https://github.com/jenkins-infra/jenkins-usage-stats[recently revised by Andrew Bayer],
+which enable us to quantify past and present trends.
+
+=== Data
+
+image:/images/post-images/2022-06-require-java-11/jvms.png[JVMS by Date]
+
+=== Analysis
+
+The above data reveals that the lifecycle of each major Java version is clearly a Gaussian function.
+Each major version of Java is released by the vendor,
+begins a phase of rapidly growing adoption,
+reaches a usage peak,
+loses support from the Jenkins project,
+begins a decline in usage,
+declines in usage to a terminal state,
+and finally reaches vendor end-of-life (EOL),
+in roughly that order.
+
+[cols="1h,1,1,1,1"]
```

Comment:
not sure what to do about it but this table breaks the display of the blog post on mobile
<img width="465" alt="image" src="https://user-images.githubusercontent.com/21194782/176021357-96f18fc7-daf8-4f71-a39a-70bff0fc1d86.png">

Reply:
Sounds like a bug in the stylesheet.

Reply:
Possibly although it’s more the number of columns. 

Would it be possible to drop Java 6?

see also https://www.uxmatters.com/mt/archives/2020/07/designing-mobile-tables.php